app/Models/Multiplayer/Room.php

```diff
@@ -376,12 +376,20 @@ public function difficultyRange()
         $extraQuery = true;
 
         if ($this->relationLoaded('playlist')) {
+            $playlistItems = $this->playlist;
+            $activePlaylistItems = $playlistItems->where('expired', '=', false);
+
             if ($this->playlist->count() > 0) {
                 $firstItem = $this->playlist[0];
+                $anyNonExpiredItems = $activePlaylistItems->isNotEmpty();
 
                 if ($firstItem->relationLoaded('beatmap')) {
                     $extraQuery = false;
                     foreach ($this->playlist as $item) {
+                        if ($anyNonExpiredItems && $item->expired) {
+                            continue;
+                        }
```

Comment:
```suggestion
                    foreach ($anyNonExipiredItems ? $activePlaylistItems : $playlistItems as $item) {
```
but maybe don't inline the conditional. and looking again, maybe
```php
...
$activePlaylistItems = ...;
// or some better named variable
$displayPlaylistItems = $activePlaylistItems->isNotEmpty() ? $activePlaylistItems : $playlistItems;
...

...
if ($extraQuery) {
    if (!isset($displayPlaylistItems)) {
        $playlistItems = ...;
        ...
        $displayPlaylistItems = ...;
    }
    // maybe short enough to be inlined
    $beatmapIds = $displayPlaylistItems->pluck('beatmap_id');
    ...
}
```